pkg/portutil/port_allocate_linux.go

```diff
@@ -25,10 +25,13 @@ import (
 
 const (
 	// This port range is compatible with Docker, FYI https://github.com/moby/moby/blob/eb9e42a09ee123af1d95bf7d46dd738258fa2109/libnetwork/portallocator/portallocator_unix.go#L7-L12
-	allocateStart = 49153
 	allocateEnd   = 60999
 )
 
+var (
+	allocateStart uint64 = 49153
+)
```

Comment:
Still conflicts if you run `nerdctl run` commands twice?

Reply:
I checked the previous commit and someone someone solved the problem of repeated executions of nerdctl run always allocating 49153 by reading the iptables nat table to determine which are the ports that have already been allocated. So using it this way here does not reintroduce the problem of always allocating the same port for multiple executions, what is solved here is the rule of port allocate when executing -p multiple times for one command

Reply:
- https://github.com/containerd/nerdctl/pull/2169

@vsiravar Could you take a llook?

Reply:
>  what is solved here is the rule of port allocate when executing -p multiple times for one command

This doesn't seem to be the case from my local test from this PR. 
```
$ sudo nerdctl run -d -p 80 -p 90 nginx
$ sudo nerdctl ps -a
CONTAINER ID    IMAGE                                       COMMAND                   CREATED               STATUS                     PORTS                                           NAMES
8d80a3e262d2    docker.io/library/nginx:latest              "/docker-entrypoint.…"    About a minute ago    Up                         0.0.0.0:49153->80/tcp, 0.0.0.0:49153->90/tcp    nginx-8d80a

Reply:
> what is solved here is the rule of port allocate when executing -p multiple times for one command

you have to rename your PR to reflect this in the title plz

Reply:
> > what is solved here is the rule of port allocate when executing -p multiple times for one command
> 
> you have to rename your PR title to reflect the above in title

Ok, I have renamed it

Reply:
> > what is solved here is the rule of port allocate when executing -p multiple times for one command
> 
> This doesn't seem to be the case from my local test from this PR.
> 
> ```
> $ sudo nerdctl run -d -p 80 -p 90 nginx
> $ sudo nerdctl ps -a
> CONTAINER ID    IMAGE                                       COMMAND                   CREATED               STATUS                     PORTS                                           NAMES
> 8d80a3e262d2    docker.io/library/nginx:latest              "/docker-entrypoint.…"    About a minute ago    Up                         0.0.0.0:49153->80/tcp, 0.0.0.0:49153->90/tcp    nginx-8d80a
> ```

Did you pull my changes and compile them, I looked and I didn't miss any changes,
I have tested locally with multiple ports and it seems to be working fine and is currently in use at work.

bfd5c6f92869   docker:dind    "/sbin/tini -- docke…"    4 seconds ago    Up        0.0.0.0:49153->7222/tcp, 0.0.0.0:49154->13022/tcp, 0.0.0.0:49155->1847/tcp, 0.0.0.0:49156->7024/tcp, 0.0.0.0:49157->8222/tcp, 0.0.0.0:49158->14022/tcp, 0.0.0.0:49159->3830/tcp, 0.0.0.0:49160->8024/tcp, 0.0.0.0:49161->9222/tcp, 0.0.0.0:49162->15022/tcp, 0.0.0.0:49163->1846/tcp, 0.0.0.0:49164->9024/tcp, 0.0.0.0:49165->10222/tcp, 0.0.0.0:49166->16022/tcp, 0.0.0.0:49167->1845/tcp, 0.0.0.0:49168->10024/tcp, 0.0.0.0:49169->3022/tcp, 0.0.0.0:49170->1848/tcp, 0.0.0.0:49171->4022/tcp, 0.0.0.0:49172->4830/tcp, 0.0.0.0:49173->61000/tcp, 0.0.0.0:49174->62000/tcp, 0.0.0.0:49175->62001/tcp, 0.0.0.0:49176->62002/tcp, 0.0.0.0:49177->62003/tcp, 0.0.0.0:49178->3255/tcp   boards_1_mh2roh_root

Reply:
![image](https://github.com/containerd/nerdctl/assets/9974436/3d1066e2-a3a5-48ee-93b9-81850c83354c)